api/filters/fieldspec/fieldspec_test.go

```diff
@@ -63,6 +63,9 @@ xxx:
 apiVersion: foo/v1
 kind: Bar
 xxx:
+metadata:
+  annotations:
+    internal.config.k8s.io/annotations-migration-resource-id: '0'
```

Comment:
@KnVerey this is something we should discuss before the next release of the `api` module to decide if we are okay with the change. 

What's happening is to migrate the annotations, kyaml/kio adds a new annotation to help keep track of the migration, then runs the filter, then removes the annotation. In the case that the filter errors, it includes in its error message the contents of the failing RNode. Unfortunately that means that this annotation gets included in the error message before we can remove it.

Because this is blocking kpt and doesn't show up in the output of _kustomize_ (we've added the annotation to kustomize's `BuildAnnotations` so that it will get removed), I'm approving this PR to go in so that they can use an unreleased version of `kyaml` until we have a better solution/decision. 

No changes need to be made if we decide we're okay with these annotations showing up in the output of `api` and `cmd/config` tests.